src/libaktualizr/primary/reportqueue_test.cc

```diff
@@ -110,16 +123,52 @@ TEST(ReportQueue, FailureRecovery) {
 
   size_t num_events = 10;
   auto http = std::make_shared<HttpFakeRq>(temp_dir.Path(), num_events);
-  ReportQueue report_queue(config, http);
+  auto sql_storage = std::make_shared<SQLStorage>(config.storage, false);
+  ReportQueue report_queue(config, http, sql_storage);
 
-  for (int i = 0; i < 10; ++i) {
+  for (size_t i = 0; i < num_events; ++i) {
     report_queue.enqueue(std_::make_unique<EcuDownloadCompletedReport>(
         Uptane::EcuSerial("FailureRecovery" + std::to_string(i)), "", true));
   }
 
-  // Wait at most 30 seconds for the messages to get processed.
+  // Wait at most 20 seconds for the messages to get processed.
+  http->expected_events_received.get_future().wait_for(std::chrono::seconds(20));
+  EXPECT_EQ(http->events_seen, num_events);
+}
+
+/* Test event store. */
```

Comment:
This test looks really good now! My last suggestion is to make the comment here more descriptive. Maybe something like "Test persistent storage of unsent events in the database across ReportQueue instantiations."